### PR TITLE
Add Nexus links tests

### PR DESCRIPTION
**Pull request by rodrigozhou**
## What was changed

Add Nexus links tests

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: test/nexus_test.go [hunk 1]
```diff
@@ -517,6 +555,72 @@ func TestAsyncOperationFromWorkflow(t *testing.T) {
...
+		require.Equal(t, tc.testConfig.Namespace, link.GetWorkflowEvent().GetNamespace())
+		require.Equal(t, handlerWfID, link.GetWorkflowEvent().GetWorkflowId())
+		require.NotEmpty(t, link.GetWorkflowEvent().GetRunId())
+		requireProtoEqual(
+			t,
+			&common.Link_WorkflowEvent_EventReference{
+				EventType: enums.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED,
+			},
+			link.GetWorkflowEvent().GetEventRef())
+		handlerRunID := link.GetWorkflowEvent().GetRunId()
```
Comment (bergundy) [inline test/nexus_test.go:588]: nit: you could have just used `requireProtoEqual` on the entire struct.

File: test/nexus_test.go [hunk 2]
```diff
@@ -290,10 +298,22 @@ func TestNexusWorkflowRunOperation(t *testing.T) {
 
 	nc := tc.newNexusClient(t, service.Name)
 
+	link := &common.Link_WorkflowEvent{
```
Comment (bergundy) [inline test/nexus_test.go:301]: This is redundant IMHO since we're already testing adding links from a workflow.

**Review by bergundy — APPROVED** [review]
I missed this in the last PR but please mark these functions as experimental in a different PR since this one has to remain a draft until we have a CLI with server 1.26.
https://github.com/temporalio/sdk-go/blob/master/temporalnexus/link_converter.go#L65-L89

File: test/nexus_test.go (hunk 1 above)
Comment (rodrigozhou) [reply test/nexus_test.go:588]: I cannot because I don't have the handler run id.

File: test/nexus_test.go [hunk 3]
```diff
@@ -31,6 +31,7 @@ import (
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
```
Comment (cretz) [inline test/nexus_test.go:34]: Is it worth a new test dependency just to show failures in this one test that should not happen anyways? I think https://pkg.go.dev/google.golang.org/protobuf/proto#Equal should be good enough for this case right?

Comment (rodrigozhou) [reply test/nexus_test.go]: Removed

**Review by cretz — APPROVED** [review]

**Pull request merged**